### ISSUE TITLE
org-babel-execute:jupyter-python: Wrong type argument: jupyter-org-request

### ISSUE THREAD

**Issue by vale981** [dzop/emacs-jupyter#234]
Hi  all,

When I'm executing a code block in org-babel for the first time, I always get this error message. All subsequent executions work just fine.

```elisp
org-babel-execute:jupyter-python: Wrong type argument: jupyter-org-request, #s(jupyter-request "acf50f27-134c-4952-b4dc-9f274d44c6f8" (24196 23680 971640 467000) t (:header (message-part "{\"msg_id\":\"5839f945-e57ab831990151b27f849e12_13\",\"msg_type\":\"status\",\"username\":\"hiro\",\"session\":\"5839f945-e57ab831990151b27f849e12\",\"date\":\"2020-04-01T09:18:57.828633Z\",\"version\":\"5.3\"}" (:msg_id "5839f945-e57ab831990151b27f849e12_13" :msg_type :status :username "hiro" :session "5839f945-e57ab831990151b27f849e12" :date "2020-04-01T09:18:57.828633Z" ...)) :msg_id "5839f945-e57ab831990151b27f849e12_13" :msg_type :status :parent_header (message-part "{\"msg_id\":\"acf50f27-134c-4952-b4dc-9f274d44c6f8\",\"msg_type\":\"execute_request\",\"version\":\"5.3\",\"username\":\"hiro\",\"session\":\"e619916e-7669-4933-a1a5-af3a5c831750\",\"date\":\"2020-04-01T11:18:56.971939+02:00\"}" (:msg_id "acf50f27-134c-4952-b4dc-9f274d44c6f8" :msg_type :execute-request :version "5.3" :username "hiro" :session "e619916e-7669-4933-a1a5-af3a5c831750" ...)) :metadata (message-part "{}" nil) ...) nil ((:status . #[128 "ÀÂ\"ÀÁ\"" [apply #[257 "ÁÀ!\0 " [jupyter-message-status-idle-p ((:header (message-part "{\"msg_id\":\"5839f945-e57ab831990151b27f849e12_13\",\"msg_type\":\"status\",\"username\":\"hiro\",\"session\":\"5839f945-e57ab831990151b27f849e12\",\"date\":\"2020-04-01T09:18:57.828633Z\",\"version\":\"5.3\"}" ...) :msg_id "5839f945-e57ab831990151b27f849e12_13" :msg_type :status :parent_header (message-part "{\"msg_id\":\"acf50f27-134c-4952-b4dc-9f274d44c6f8\",\"msg_type\":\"execute_request\",\"version\":\"5.3\",\"username\":\"hiro\",\"session\":\"e619916e-7669-4933-a1a5-af3a5c831750\",\"date\":\"2020-04-01T11:18:56.971939+02:00\"}" ...) :metadata (message-part "{}" nil) ...))] 4 "
```

**Comment by vale981** [comment]
never mind. this is somehow related to poly-mode....

**Comment by veprbl** [comment]
I have the same issue with a slightly different message:
```
or: Wrong type argument: jupyter-org-request, #s(jupyter-request "66d603a6-6ef6-492b-834b-3580c20dd533" (24196 59341 869477 0) nil nil nil nil)
```

**Comment by vale981** [comment]
Then let's reopen this.

**Comment by dzop** [comment]
Yes there may be issues with the recent update, I somehow managed to push almost done, but not quite, commits that I was working on into the `master` branch a couple days ago. I was hoping I could fix the issues before anyone notices, but it doesn't look that way. I just reverted back to b691d38483b6540d42d482a32d35eb54178e5658. Could you update to the most recent master a4e0616ed4acd90f1ee7cb981bb8fb6bc05712d6 to see if the problem goes away?

**Comment by veprbl** [comment]
@dzop 
Unfortunately the problem is still occurring:
```
Error running timer ‘jupyter-event-handler’: (error "Cell for request not found")
Wrong type argument: jupyter-org-request, #s(jupyter-request "ac63c4f9-65d8-4350-83ce-ad5353834a92" (24198 11706 720705 0) t (:header (message-part "{\"msg_id\":\"aacfa1a4-89ebce9ff42aac2458acec73\",\"msg_type\":\"execute_reply\",\"username\":\"veprbl\",\"session\":\"340e7b7f-191fdcdd136fa641654b5060\",\"date\":\"2020-04-02T18:23:54.731760Z\",\"version\":\"5.3\"}" (:msg_id "aacfa1a4-89ebce9ff42aac2458acec73" :msg_type :execute-reply :username "veprbl" :session "340e7b7f-191fdcdd136fa641654b5060" :date "2020-04-02T18:23:54.731760Z" ...)) :msg_id "aacfa1a4-89ebce9ff42aac2458acec73" :msg_type :execute-reply :parent_header (message-part "{\"msg_id\":\"ac63c4f9-65d8-4350-83ce-ad5353834a92\",\"msg_type\":\"execute_request\",\"version\":\"5.3\",\"username\":\"veprbl\",\"session\":\"581ba4ea-d01f-4161-b1a7-b1bb8030e620\",\"date\":\"2020-04-02T14:23:54.720907-04:00\"}" (:msg_id "ac63c4f9-65d8-4350-83ce-ad5353834a92" :msg_type :execute-request :version "5.3" :username "veprbl" :session "581ba4ea-d01f-4161-b1a7-b1bb8030e620" ...)) :metadata (message-part "{\"started\":\"2020-04-02T18:23:54.722308Z\",\"dependencies_met\":true,\"engine\":\"a10d3638-8b9e-4990-9ac2-508c6708e22b\",\"status\":\"ok\"}" nil) ...) nil ((:status . #[128 "ÀÂ\"ÀÁ\"" [apply (closure ((msg :header (message-part "{\"msg_id\":\"f80b73b7-8323de8434ff84b03987bf58\",\"msg_type\":\"status\",\"username\":\"veprbl\",\"session\":\"340e7b7f-191fdcdd136fa641654b5060\",\"date\":\"2020-04-02T18:23:54.732362Z\",\"version\":\"5.3\"}" (:msg_id "f80b73b7-8323de8434ff84b03987bf58" :msg_type :status :username "veprbl" :session "340e7b7f-191fdcdd136fa641654b5060" :date "2020-04-02T18:23:54.732362Z" ...)) :msg_id "f80b73b7-8323de8434ff84b03987bf58" :msg_type :status :parent_header (message-part "{\"msg_id\":\"ac63c4f9-65d8-4350-83ce-ad5353834a92\",\"msg_type\":\"execute_request\",\"version\":\"5.3\",\"username\":\"veprbl\",\"session\":\"581ba4ea-d01f-4161-b1a7-b1bb8030e620\",\"date\":\"2020-04-02T14:23:54.720907-04:00\"}" (:msg_id "ac63c4f9-65d8-4350-83ce-ad5353834a92" :msg_type :execute-request :version "5.3" :username "veprbl" :session "581ba4ea-d01f-4161-b1a7-b1bb8030e620" ...)) :metadata ...) (progress-msg) (timeout . 10) (cb . jupyter-message-status-idle-p) (msg-type . :status) (req . #0) t) (m) (setq msg (if (funcall cb m) (progn m)))) identity nil] 5 nil])))
```
The MRE (based on @dickmao 's https://github.com/dzop/emacs-jupyter/issues/176#issuecomment-607775021):
```
emacs -Q --batch --eval "(require 'jupyter)" --eval "(setq org-babel-load-languages (quote ((jupyter . t))))" --eval "(setq org-confirm-babel-evaluate nil)" --eval "(with-temp-buffer (org-mode) (insert \"#+BEGIN_SRC jupyter-python :session py\n3.14\n#+END_SRC\n\") (princ (buffer-string)) (call-interactively #'org-ctrl-c-ctrl-c))"
```
I've bisected it to find that e6e9694f9435d2e67f1053e10712fefe58d53ab9 is the first bad commit.

**Comment by dzop** [comment]
@veprbl Thanks, I was able to reproduce the issue. Should be fixed by 5c2469dbe40a3ee82537ecb549ee0b05ef35e194.

https://github.com/dzop/emacs-jupyter/commit/e6e9694f9435d2e67f1053e10712fefe58d53ab9 was wrong to do in the first place. Messages are handled somewhat asynchronously and the problem seems to be that `org-babel-jupyter-current-src-block-params` was being set to `nil` sometime before some of the messages that needed it to be non-nil were generated, see [jupyter-generate-request](https://github.com/dzop/emacs-jupyter/blob/5c2469dbe40a3ee82537ecb549ee0b05ef35e194/jupyter-org-client.el#L119).
